src/sql/base/query/browser/untitledQueryEditorInput.ts

```diff
@@ -44,6 +48,28 @@ export class UntitledQueryEditorInput extends QueryEditorInput implements IUntit
 		return this.text.model.hasAssociatedFilePath;
 	}
 
+	override async save(group: GroupIdentifier, options?: ISaveOptions): Promise<IEditorInput | undefined> {
```

Comment:
there are room for refactoring so that you don't have to repeat the same comment